test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowableJSTest.scala

```diff
@@ -0,0 +1,35 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013--2018, LAMP/EPFL  **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.testsuite.javalib.lang
+
+import scala.scalajs.js
+import scala.scalajs.LinkingInfo.assumingES6
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+class ThrowableJSTest {
```

Comment:
We seem to not have tests that test catching behavior (i.e. interaction between Scala throwables and `js.JavaScriptException`). I think it would be a good idea to add some of them before this change to make sure we do not alter any behavior.

My current assumption is that `js.JavaScriptException` catches anything as `js.Any` so this change would not alter the behavior. Then again this would be dangerous, since it would basically mean there is no easy way to catch JS throwables without catching fatal exceptions (notably `UndefinedBehaviorError`). But than that would mean the `js.JavaScriptException` would only catch non-throwables, which would in turn be broken by this change (IIUC).

Reply:
Ah, right. We test the other way around (we test for `Throwable` not `not Error`, obvious in hindsight since JS can throw anything). So yes, this shouldn't break anything (but we probably still should have tests).